### ISSUE TITLE
Should update translation to use 0.17 pulumi features.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [pulumi/tf2pulumi#85]
Improvements include:
1. we can use `pulumi.interpolate` if we're doing string concatenations.
2. we can use proxies/lifted-props to replace suitable `.apply` calls on a single output.

**Comment by CyrusNajmabadi** [comment]
I was hoping i could do this in a short amount of time, but i leaked out of my timebox comfort zone.  @pgavlin could likely do it in seconds when there is spare time :)   This will def help clean up what we produce. For example, we would go from:

```ts
 * const certValidation = new aws.route53.Record("cert_validation", {
 *     records: [certCertificate.domainValidationOptions.apply(domainValidationOptions => domainValidationOptions[0].resourceRecordValue)],
 *     ttl: 60,
 *     type: certCertificate.domainValidationOptions.apply(domainValidationOptions => domainValidationOptions[0].resourceRecordType),
 *     zoneId: zone.apply(zone => zone.id),
 * });
```

Into:

```ts
 * const certValidation = new aws.route53.Record("cert_validation", {
 *     records: [certCertificate.domainValidationOptions[0].resourceRecordValue],
 *     ttl: 60,
 *     type: certCertificate.domainValidationOptions[0].resourceRecordType,
 *     zoneId: zone.id,
 * });
```

**Comment by pgavlin** [comment]
Partially addressed in #103.

**Comment by pgavlin** [comment]
Remaining work includes recognizing index expressions and lowering these to proxy applies as well.
